### ISSUE TITLE
Entering <space> in variable name editor shouldn't close the editor

### ISSUE THREAD

**Issue by Geoffree3** [Microsoft/pxt#3203]
Entering a variable name containing a space is difficult as the name editor closes straight after pressing <space>!   However, the space is there when you open it again, so this is rather a strange behaviour and inconsistent.  It does allow you to add another word to the end (and hence have a space in the variable name) but it's a awkward way to achieve this. There is an 'OK button' to press when done, so why close on <space> anyway?

Google Blockly is OK (on the same Chrome browser) - 
<img width="527" alt="blockly varaible" src="https://user-images.githubusercontent.com/3199856/31469677-8bb8987e-aeda-11e7-87fa-7d25e5c1a72b.png">

**Comment by abchatra** [comment]
Our backend is typescript and we don't support space in the variable name. This is by design.

**Comment by Geoffree3** [comment]
OK, then why allow users to enter a <space>?  As I said, the <space> is appended to the variable name and upon rename, users can type another word and hence have a space in the variable name.  I suggest fixing to ignore <space> characters

**Comment by Geoffree3** [comment]
{space within arrow brackets doesn't display}  OK, then why allow users to enter a space? As I said, the spaceis appended to the variable name and upon rename, users can type another word and hence have a space in the variable name. I suggest fixing to ignore characters

**Comment by carlosperate** [comment]
I would argue that the ticket is valid in the sense that having the modal close itself when the space key is pressed is a bit of an unexpected reaction.
Not only that, but it creates a variable name with the first word, there is no indication of what happened or why it happened, and so I found it slightly confusing the first time I encountered it.

Here is an example of a user would experience when trying to create a variable named `my variable`:

![my_var](https://user-images.githubusercontent.com/4189262/31640772-3f2fc0f2-b2d8-11e7-89a6-7d5cb430823d.gif)

Blockly gets around this issue by replacing the spaces, and passing the variable name through a "unique name generator" function. I'm not saying MakeCode should do the same, but it would be useful if it was clear that spaces are not permitted.
